src/core/core.scale.js

```diff
@@ -668,16 +676,11 @@ module.exports = function(Chart) {
 						context.lineTo(itemToDraw.tx2, itemToDraw.ty2);
 					}
 
-					if (gridLines.drawOnChartArea) {
-						context.moveTo(itemToDraw.x1, itemToDraw.y1);
-						context.lineTo(itemToDraw.x2, itemToDraw.y2);
-					}
-
 					context.stroke();
 					context.restore();
 				}
 
-				if (optionTicks.display) {
```

Comment:
Hmmm, for grid lines we have the following logic:
if label is a string, draw label, tick and grid line
if label is `null` or `undefined` hide the label, tick and gridline.

This allows users to filter out grid lines by return `null` from the tick callback. [here](https://github.com/chartjs/Chart.js/blob/master/samples/scales/filtering-labels.html) is an example of filtering